src/components/layout/UserPortalUpdatePrompts.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * @author aramsey
+ *
+ * The components for prompting users to update their user account information
+ * within the User Portal if necessary.
+ *
+ * The Snackbar is used when the user is within the grace period to update
+ * their information.
+ *
+ * The dialog is used when the grace period has expired and the user must
+ * update their information in order to use CyVerse services.
+ */
+
+import React, { useEffect, useState } from "react";
+
+import { build } from "@cyverse-de/ui-lib";
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    IconButton,
+    makeStyles,
+    Snackbar,
+    Typography,
+    useTheme,
+} from "@material-ui/core";
+import { Close } from "@material-ui/icons";
+import { Alert, AlertTitle } from "@material-ui/lab";
+
+import constants from "constants.js";
+import { useUserProfile } from "contexts/userProfile";
+import { useTranslation } from "i18n";
+import ids from "./ids";
+import { usePortalStatus } from "serviceFacades/users";
+import styles from "./styles";
+
+const useStyles = makeStyles(styles);
+
+function UserPortalUpdatePrompts() {
+    const [userProfile] = useUserProfile();
+
+    const { t } = useTranslation("common");
+    const theme = useTheme();
+    const classes = useStyles();
+
+    const baseId = ids.USER_PORTAL_UPDATE_DLG;
+    const dialogTitleId = build(baseId, ids.DIALOG_TITLE);
+
+    const [showSnackbar, setShowSnackbar] = useState(false);
+
+    const { data: userPortalResp } = usePortalStatus(userProfile?.id, (err) => {
+        console.log("Received error response from user portal API", err);
+    });
+
+    useEffect(() => {
+        setShowSnackbar(userPortalResp?.warning_required);
+    }, [userPortalResp]);
+
+    const onCloseAnnouncer = () => setShowSnackbar(false);
+
+    const onUpdateNow = () => {
+        const updateUrl = userPortalResp?.update_url;
+        if (updateUrl) {
+            const portalUrl = `${updateUrl}&redirectUrl=${encodeURIComponent(
+                window.location.href
+            )}`;
+            window.location.replace(portalUrl);
```

Comment:
I didn't catch this until I tried it after this PR was merged, but a minor inconvenience of using `replace` here is that it breaks the user's browser history, since the user portal simply redirects back and does not also use replace.

So which ever page the user was viewing when they get the dialog will be replaced with the user portal URL after they are redirected back to the DE.

Reply:
Almost forgot about this comment! 😅 
I don't think I'm seeing what you're seeing though, or maybe I'm misunderstanding.  When I tested it, I:
* Navigated to de.cyverse.org
* Got the pop-up to update
* Got sent to the user portal, updated my info, hit "Proceed to application"
* Got redirected back to the DE with de.cyverse.org as my url
* Back button takes me to user portal, back button again takes me to the DE